File: app/src/main/java/com/example/shop/dao/UserDao.java
package com.example.shop.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.shop.entity.User;

import java.util.List;

@Dao
public interface UserDao {
    @Insert
    void insertUser(User...users);
    @Update
    void updateUser(User...users);
    @Query("SELECT * FROM USER WHERE NAME =:name")
    User getUser(int name);
    @Query("SELECT * FROM USER")
    List<User> getAllUser();
    @Query("SELECT password FROM USER WHERE PHONENUMBER=:phoneNumber")
    String getPassword(String phoneNumber);
    @Query("SELECT * FROM USER WHERE PHONENUMBER=:phoneNumber")
    User getUser(String phoneNumber);
}

File: app/src/main/java/com/example/shop/dao/UserDao.kt
package com.example.shop.dao

import androidx.room.*
import com.example.shop.entity.User

@Dao
interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertUser(vararg users: User?)

    @Query("SELECT * FROM User WHERE phoneNumber=:phoneNumber LIMIT 1")
    suspend fun getUser(phoneNumber: String): User?

    @Query("SELECT name FROM User WHERE phoneNumber=:phoneNumber LIMIT 1")
    suspend fun getName(phoneNumber: String): String?

    @Query("SELECT password FROM User WHERE phoneNumber=:phoneNumber LIMIT 1")
    suspend fun getPassword(phoneNumber: String): String?
}